test/test_numpy_interop.py

```diff
@@ -481,6 +481,15 @@ def test_numpy_scalar_cmp(self, device, dtype):
                     self.assertFalse(t == a)
                 else:
                     self.assertTrue(t == a)
+    @onlyCPU
+    def test_empty_tensors_interop(self, device):
+        x = torch.empty((8, 0), dtype=torch.float16)
+        y = torch.from_numpy(np.empty((0, 8, 0), dtype=np.float16))
```

Comment:
Note that if you want to create this without numpy, you can do `x = torch.empty_strided((0, 8, 0), (0, 0, 0))`.